# The Best Places to Eat Italian throughout the US

Coursera IBM Data Science Professional Certificate Capstone

By: Kristianne Baugh

## Table of Contents

-Introduction/Business Problem

-Data 

-Methodology

-Analysis

-Results and Discussion

-Conclusion

# 1. Introduction/Business Problem

## Background

According to the Daily Meal webblog, there are no shortages of Italian Restaaurants to choose from throughout the United States. Each state has the best Italian restaurant, by popular votes. https://www.thedailymeal.com/eat/best-italian-restaurant-every-state-gallery

Also, the Daily Meal webblog has an article about the top 50 best Italian restaurants in the United states, and many of those restaurants are in New York. https://www.thedailymeal.com/eat/americas-best-italian-restaurants-gallery

However, what if I wanted a list of the best Italian restaurants in certain areas, such as New York, Michigan, Houston, Seattle, and Massachusetts? Also, what if I wanted to compare them? 

## Problem Description

While the Daily Meal has a great list of places to visit, if a tourist is craving Italian food, my approach to this problem will be to use Foursquare API to collect the data on Italian restaurants within New York, Michigan, Houston, Seattle, and Massachusetts. I will then compare them in order to determine the best places to eat Italian food. 

## Target Audience

My target audience will consist of mainly tourists who are looking for authentic Italian restaurants throughout New York, Michigan, Houston, Seattle, and Massachusetts. 

## Success Criteria

The success of this investigation will depend upon the following: 

1. Popularity of the restaurant
2. Price range of the restaurant
3. Wait time of the restaurant

# 2. Data Selection

I plan on using the Foursquare API to collect data about the locations of pasta restaurants in the following US cities: New York, Michigan, Houston, Seattle, and Massachusetts. I am hoping that this data will provide enough information to determine the best restaurant to visit according to the above-mentioned success criteria.

I will use Foursquare to gather the top Italian restuarants within those cities, and then compare their popularity, prices, and wait time in order to determine the best restaurants in those locations. I will then compare the top restaurants with each other to determine which one would be the absolute best Italian restaurant to visit. 

In [15]:
import pandas as pd
import locale
import requests
import numpy as np
import matplotlib.cm as cm
import matplotlib.colors as colors
import matplotlib.pyplot as plt
import json
from math import sin, cos, sqrt, atan2, radians
from sklearn.cluster import KMeans
import matplotlib.path as mpltPath
from geopy.geocoders import Nominatim
from pandas.io.json import json_normalize

locale.setlocale(locale.LC_ALL, 'en_US.UTF8')
pd.set_option('display.max_column',None)
pd.set_option('display.max_rows', None)
pd.set_option('display.max_colwidth', -1)

/opt/conda/envs/Python-3.7-main/lib/python3.7/site-packages/ipykernel/__main__.py:18: FutureWarning: Passing a negative integer is deprecated in version 1.0 and will not be supported in future version. Instead, use None to not limit the column width.


In [21]:
!python3 -m pip install folium
import folium

In [7]:
CLIENT_ID = 'BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ'
CLIENT_SECRET = 'GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I'
ACCESS_TOKEN = 'BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV'
VERSION = '20180604'
LIMIT = 30
print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ
CLIENT_SECRET:GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I


# 3. Methodology

## Let's say that I am in New York City and Staying at The Plaza Hotel. 

In [10]:
address = '5th Avenue, New York, NY'

geolocator = Nominatim(user_agent="foursquare_agent")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)

40.8152452 -73.9357728


I'm looking for the best Italian Restaurant in New York City right now.

In [11]:
search_query = 'Italian'
radius = 1000
print(search_query + ' .... Got it!')

Italian .... Got it!


In [12]:
url = 'https://api.foursquare.com/v2/venues/search?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&ll=40.8152452,-73.9357728&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604&query=Italian&radius=1000&limit=30'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude,ACCESS_TOKEN, VERSION, search_query, radius, LIMIT)
url

'https://api.foursquare.com/v2/venues/search?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&ll=40.8152452,-73.9357728&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604&query=Italian&radius=1000&limit=30'

In [13]:
results = requests.get(url).json()
results

{'meta': {'code': 200, 'requestId': '5ffba0bcf5b91f3347efaeff'},
 'notifications': [{'type': 'notificationTray', 'item': {'unreadCount': 0}}],
 'response': {'venues': [{'id': '4c40f185e26920a1d5bb5de7',
    'name': "Yolanda's Italian Pizzeria & Restaurant",
    'location': {'address': '292 E 149th St',
     'crossStreet': 'off Morris Avenue',
     'lat': 40.81709844985412,
     'lng': -73.92186622574978,
     'labeledLatLngs': [{'label': 'display',
       'lat': 40.81709844985412,
       'lng': -73.92186622574978},
      {'label': 'entrance', 'lat': 40.816997, 'lng': -73.921803}],
     'distance': 1189,
     'postalCode': '10451',
     'cc': 'US',
     'city': 'Bronx',
     'state': 'NY',
     'country': 'United States',
     'formattedAddress': ['292 E 149th St (off Morris Avenue)',
      'Bronx, NY 10451']},
    'categories': [{'id': '4bf58dd8d48988d1ca941735',
      'name': 'Pizza Place',
      'pluralName': 'Pizza Places',
      'shortName': 'Pizza',
      'icon': {'prefix': 'https

Results in New York

In [16]:
venues = results['response']['venues']

dataframe = json_normalize(venues)
dataframe.head()

/opt/conda/envs/Python-3.7-main/lib/python3.7/site-packages/ipykernel/__main__.py:3: FutureWarning: pandas.io.json.json_normalize is deprecated, use pandas.json_normalize instead
  app.launch_new_instance()


id                                     name  \
0  4c40f185e26920a1d5bb5de7  Yolanda's Italian Pizzeria & Restaurant   
1  4e74bc8ca809582dd59b548c  Passarelli's Italian Sausage              

                                                                                                                                                                                                                     categories  \
0  [{'id': '4bf58dd8d48988d1ca941735', 'name': 'Pizza Place', 'pluralName': 'Pizza Places', 'shortName': 'Pizza', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/pizza_', 'suffix': '.png'}, 'primary': True}]   
1  []                                                                                                                                                                                                                             

     referralId  hasPerk  location.address location.crossStreet  location.lat  \
0  v-1610326204  False    292 E 149th St    off Morris Avenue    40.817098      
1  v-1610326204  False    Randall's Island  NaN                  40.814252      

   location.lng  \
0 -73.921866      
1 -73.934211      

                                                                                                                   location.labeledLatLngs  \
0  [{'label': 'display', 'lat': 40.81709844985412, 'lng': -73.92186622574978}, {'label': 'entrance', 'lat': 40.816997, 'lng': -73.921803}]   
1  [{'label': 'display', 'lat': 40.81425232316303, 'lng': -73.93421114826005}]                                                               

   location.distance location.postalCode location.cc location.city  \
0  1189               10451               US          Bronx          
1  171                NaN                 US          Bronx          

  location.state location.country  \
0  NY             United States     
1  NY             United States     

                               location.formattedAddress  
0  [292 E 149th St (off Morris Avenue), Bronx, NY 10451]  
1  [Randall's Island, Bronx, NY]

So it appears that there are only 2 Italian Restaurants within New York City that are close to The Plaza Hotel. 

Now, I want to clean up the NY dataframe :

In [17]:
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
NY_dataframe_filtered = dataframe.loc[:, filtered_columns]

def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']


NY_dataframe_filtered['categories'] = NY_dataframe_filtered.apply(get_category_type, axis=1)

NY_dataframe_filtered.columns = [column.split('.')[-1] for column in NY_dataframe_filtered.columns]

NY_dataframe_filtered

name   categories           address  \
0  Yolanda's Italian Pizzeria & Restaurant  Pizza Place  292 E 149th St     
1  Passarelli's Italian Sausage             None         Randall's Island   

         crossStreet        lat        lng  \
0  off Morris Avenue  40.817098 -73.921866   
1  NaN                40.814252 -73.934211   

                                                                                                                            labeledLatLngs  \
0  [{'label': 'display', 'lat': 40.81709844985412, 'lng': -73.92186622574978}, {'label': 'entrance', 'lat': 40.816997, 'lng': -73.921803}]   
1  [{'label': 'display', 'lat': 40.81425232316303, 'lng': -73.93421114826005}]                                                               

   distance postalCode  cc   city state        country  \
0  1189      10451      US  Bronx  NY    United States   
1  171       NaN        US  Bronx  NY    United States   

                                        formattedAddress  \
0  [292 E 149th St (off Morris Avenue), Bronx, NY 10451]   
1  [Randall's Island, Bronx, NY]                           

                         id  
0  4c40f185e26920a1d5bb5de7  
1  4e74bc8ca809582dd59b548c

Get a nice visualization

In [22]:
NY_dataframe_filtered.name

NY_venues_map = folium.Map(location=[latitude, longitude], zoom_start=13)

folium.CircleMarker(
    [latitude, longitude],
    radius=10,
    color='red',
    popup='The Plaza Hotel',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(NY_venues_map)


for lat, lng, label in zip(NY_dataframe_filtered.lat, NY_dataframe_filtered.lng, NY_dataframe_filtered.categories):
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        popup=label,
        fill = True,
        fill_color='blue',
        fill_opacity=0.6
    ).add_to(NY_venues_map)

NY_venues_map

## I will now do the same for Michigan, particulary Detroit. Let's say that I visited the Detroit Institute of Arts and I am, of course, craving Italian food. 

In [23]:
address = '5200 Woodward Avenue, Detroit, Michigan'

geolocator = Nominatim(user_agent="foursquare_agent")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)

42.3595105 -83.06452586249985


In [24]:
search_query = 'Italian'
radius = 1000
print(search_query + ' .... Got it!')

Italian .... Got it!


In [25]:
url = 'https://api.foursquare.com/v2/venues/search?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&ll=42.3595105,-83.06452586249985&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604&query=Italian&radius=1000&limit=30'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude,ACCESS_TOKEN, VERSION, search_query, radius, LIMIT)
url

'https://api.foursquare.com/v2/venues/search?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&ll=42.3595105,-83.06452586249985&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604&query=Italian&radius=1000&limit=30'

In [26]:
results = requests.get(url).json()
results

venues = results['response']['venues']

dataframe = json_normalize(venues)
dataframe.head()

/opt/conda/envs/Python-3.7-main/lib/python3.7/site-packages/ipykernel/__main__.py:6: FutureWarning: pandas.io.json.json_normalize is deprecated, use pandas.json_normalize instead


id     name  \
0  4bce2b0acc8cd13aaee3c3cf  Mario's   

                                                                                                                                                                                                                                       categories  \
0  [{'id': '4bf58dd8d48988d110941735', 'name': 'Italian Restaurant', 'pluralName': 'Italian Restaurants', 'shortName': 'Italian', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/italian_', 'suffix': '.png'}, 'primary': True}]   

     referralId  hasPerk location.address                location.crossStreet  \
0  v-1610326306  False    4222 2nd Ave     btwn W.Canfield St. & W.Willis St.   

   location.lat  location.lng  \
0  42.350318    -83.065788      

                                                                                                                   location.labeledLatLngs  \
0  [{'label': 'display', 'lat': 42.35031768732882, 'lng': -83.06578768571968}, {'label': 'entrance', 'lat': 42.350334, 'lng': -83.065915}]   

   location.distance location.postalCode location.cc location.city  \
0  1028               48201               US          Detroit        

  location.state location.country  \
0  MI             United States     

                                                location.formattedAddress  \
0  [4222 2nd Ave (btwn W.Canfield St. & W.Willis St.), Detroit, MI 48201]   

  delivery.id  \
0  2038542      

                                                                                                                                                                                          delivery.url  \
0  https://www.grubhub.com/restaurant/marios-restaurant-4222-2nd-ave-detroit/2038542?affiliate=1131&utm_source=foursquare-affiliate-network&utm_medium=affiliate&utm_campaign=1131&utm_content=2038542   

  delivery.provider.name             delivery.provider.icon.prefix  \
0  grubhub                https://fastly.4sqi.net/img/general/cap/   

  delivery.provider.icon.sizes              delivery.provider.icon.name  \
0  [40, 50]                     /delivery_provider_grubhub_20180129.png   

  venuePage.id  
0  56970947

In [27]:
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
MI_dataframe_filtered = dataframe.loc[:, filtered_columns]

def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']


MI_dataframe_filtered['categories'] = MI_dataframe_filtered.apply(get_category_type, axis=1)

MI_dataframe_filtered.columns = [column.split('.')[-1] for column in MI_dataframe_filtered.columns]

MI_dataframe_filtered

name          categories       address  \
0  Mario's  Italian Restaurant  4222 2nd Ave   

                          crossStreet        lat        lng  \
0  btwn W.Canfield St. & W.Willis St.  42.350318 -83.065788   

                                                                                                                            labeledLatLngs  \
0  [{'label': 'display', 'lat': 42.35031768732882, 'lng': -83.06578768571968}, {'label': 'entrance', 'lat': 42.350334, 'lng': -83.065915}]   

   distance postalCode  cc     city state        country  \
0  1028      48201      US  Detroit  MI    United States   

                                                         formattedAddress  \
0  [4222 2nd Ave (btwn W.Canfield St. & W.Willis St.), Detroit, MI 48201]   

                         id  
0  4bce2b0acc8cd13aaee3c3cf

In [28]:
MI_dataframe_filtered.name

MI_venues_map = folium.Map(location=[latitude, longitude], zoom_start=13)

folium.CircleMarker(
    [latitude, longitude],
    radius=10,
    color='red',
    popup='Detroit Institute of Arts',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(MI_venues_map)


for lat, lng, label in zip(MI_dataframe_filtered.lat, MI_dataframe_filtered.lng, MI_dataframe_filtered.categories):
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        popup=label,
        fill = True,
        fill_color='blue',
        fill_opacity=0.6
    ).add_to(MI_venues_map)

MI_venues_map

# Houston, Texas - Rice University

In [29]:
address = '6100 Main Street, Houston, Texas'

geolocator = Nominatim(user_agent="foursquare_agent")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)

29.724626524096387 -95.39042063253012


In [30]:
search_query = 'Italian'
radius = 1000
print(search_query + ' .... Got it!')

Italian .... Got it!


In [31]:
url = 'https://api.foursquare.com/v2/venues/search?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&ll=29.724626524096387,-95.39042063253012&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604&query=Italian&radius=1000&limit=30'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude,ACCESS_TOKEN, VERSION, search_query, radius, LIMIT)
url

'https://api.foursquare.com/v2/venues/search?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&ll=29.724626524096387,-95.39042063253012&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604&query=Italian&radius=1000&limit=30'

In [32]:
results = requests.get(url).json()
results

venues = results['response']['venues']

dataframe = json_normalize(venues)
dataframe.head()

/opt/conda/envs/Python-3.7-main/lib/python3.7/site-packages/ipykernel/__main__.py:6: FutureWarning: pandas.io.json.json_normalize is deprecated, use pandas.json_normalize instead


id                                      name  \
0  4abb7d4ff964a520d78320e3  Italian Cultural & Community Center        
1  4f32962b19836c91c7e649f1  Trattoria Pasquale Italian Bar and Grill   

                                                                                                                                                                                                                                             categories  \
0  [{'id': '52e81612bcbc57f1066b7a32', 'name': 'Cultural Center', 'pluralName': 'Cultural Centers', 'shortName': 'Cultural Center', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/building/default_', 'suffix': '.png'}, 'primary': True}]   
1  [{'id': '4d4b7105d754a06374d81259', 'name': 'Food', 'pluralName': 'Food', 'shortName': 'Food', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/default_', 'suffix': '.png'}, 'primary': True}]                                         

     referralId  hasPerk    location.address location.crossStreet  \
0  v-1610326314  False    1101 Milford St     Bayard Lane           
1  v-1610326314  False    4412 Montrose Blvd  NaN                   

   location.lat  location.lng  \
0  29.729022    -95.392195      
1  29.733276    -95.391487      

                                                                                                                  location.labeledLatLngs  \
0  [{'label': 'display', 'lat': 29.729021999999997, 'lng': -95.392195}, {'label': 'entrance', 'lat': 29.729173, 'lng': -95.392182}]         
1  [{'label': 'entrance', 'lat': 29.733314, 'lng': -95.391383}, {'label': 'display', 'lat': 29.7332763671875, 'lng': -95.39148712158203}]   

   location.distance location.postalCode location.cc location.city  \
0  518                77006               US          Houston        
1  968                77006               US          Houston        

  location.state location.country  \
0  TX             United States     
1  TX             United States     

                            location.formattedAddress  
0  [1101 Milford St (Bayard Lane), Houston, TX 77006]  
1  [4412 Montrose Blvd, Houston, TX 77006]

In [33]:
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
TX_dataframe_filtered = dataframe.loc[:, filtered_columns]

def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']


TX_dataframe_filtered['categories'] = TX_dataframe_filtered.apply(get_category_type, axis=1)

TX_dataframe_filtered.columns = [column.split('.')[-1] for column in TX_dataframe_filtered.columns]

TX_dataframe_filtered

name       categories  \
0  Italian Cultural & Community Center       Cultural Center   
1  Trattoria Pasquale Italian Bar and Grill  Food              

              address  crossStreet        lat        lng  \
0  1101 Milford St     Bayard Lane  29.729022 -95.392195   
1  4412 Montrose Blvd  NaN          29.733276 -95.391487   

                                                                                                                           labeledLatLngs  \
0  [{'label': 'display', 'lat': 29.729021999999997, 'lng': -95.392195}, {'label': 'entrance', 'lat': 29.729173, 'lng': -95.392182}]         
1  [{'label': 'entrance', 'lat': 29.733314, 'lng': -95.391383}, {'label': 'display', 'lat': 29.7332763671875, 'lng': -95.39148712158203}]   

   distance postalCode  cc     city state        country  \
0  518       77006      US  Houston  TX    United States   
1  968       77006      US  Houston  TX    United States   

                                     formattedAddress  \
0  [1101 Milford St (Bayard Lane), Houston, TX 77006]   
1  [4412 Montrose Blvd, Houston, TX 77006]              

                         id  
0  4abb7d4ff964a520d78320e3  
1  4f32962b19836c91c7e649f1

In [34]:
TX_dataframe_filtered.name

TX_venues_map = folium.Map(location=[latitude, longitude], zoom_start=13)

folium.CircleMarker(
    [latitude, longitude],
    radius=10,
    color='red',
    popup='Rice University',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(TX_venues_map)


for lat, lng, label in zip(TX_dataframe_filtered.lat, TX_dataframe_filtered.lng, TX_dataframe_filtered.categories):
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        popup=label,
        fill = True,
        fill_color='blue',
        fill_opacity=0.6
    ).add_to(TX_venues_map)

TX_venues_map

# Seattle, Washington - Space Needle

In [35]:
address = '400 Broad St., Seattle, Washington'

geolocator = Nominatim(user_agent="foursquare_agent")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)

47.6205131 -122.34930359883187


In [36]:
search_query = 'Italian'
radius = 1000
print(search_query + ' .... Got it!')

Italian .... Got it!


In [37]:
url = 'https://api.foursquare.com/v2/venues/search?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&ll=47.6205131, -122.34930359883187&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604&query=Italian&radius=1000&limit=30'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude,ACCESS_TOKEN, VERSION, search_query, radius, LIMIT)
url

'https://api.foursquare.com/v2/venues/search?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&ll=47.6205131, -122.34930359883187&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604&query=Italian&radius=1000&limit=30'

In [38]:
results = requests.get(url).json()
results

venues = results['response']['venues']

dataframe = json_normalize(venues)
dataframe.head()

/opt/conda/envs/Python-3.7-main/lib/python3.7/site-packages/ipykernel/__main__.py:6: FutureWarning: pandas.io.json.json_normalize is deprecated, use pandas.json_normalize instead


id                                   name  \
0  50a2a4b8e4b0d2d4297f779c  WiseGuy Italian Street Food             
1  4f3228a119836c91c7bbf2cc  Cucina De Ra Italian Restaurant         
2  4f3227cd19836c91c7bba092  Merlino Italian Baking Company          
3  582fa3ae9b615c49208a81dc  Procaccini's Family Italian Sithraunt   
4  4befa9fdea570f47a59e8fd2  Calozzi's Italian Kitchen               

                                                                                                                                                                                                                                          categories  \
0  [{'id': '4bf58dd8d48988d1cb941735', 'name': 'Food Truck', 'pluralName': 'Food Trucks', 'shortName': 'Food Truck', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/streetfood_', 'suffix': '.png'}, 'primary': True}]                
1  [{'id': '4d4b7105d754a06374d81259', 'name': 'Food', 'pluralName': 'Food', 'shortName': 'Food', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/default_', 'suffix': '.png'}, 'primary': True}]                                      
2  [{'id': '4bf58dd8d48988d16a941735', 'name': 'Bakery', 'pluralName': 'Bakeries', 'shortName': 'Bakery', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/bakery_', 'suffix': '.png'}, 'primary': True}]                               
3  [{'id': '4bf58dd8d48988d11a941735', 'name': 'Other Nightlife', 'pluralName': 'Other Nightlife', 'shortName': 'Nightlife', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/nightlife/default_', 'suffix': '.png'}, 'primary': True}]      
4  [{'id': '4bf58dd8d48988d14e941735', 'name': 'American Restaurant', 'pluralName': 'American Restaurants', 'shortName': 'American', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/default_', 'suffix': '.png'}, 'primary': True}]   

     referralId  hasPerk  location.address  location.lat  location.lng  \
0  v-1610326325  False    Mobile            47.622078    -122.337258     
1  v-1610326325  False    2801 Western Ave  47.615631    -122.353310     
2  v-1610326325  False    771 Valley St     47.626080    -122.341881     
3  v-1610326325  False    NaN               47.625269    -122.357977     
4  v-1610326325  False    90 Blanchard St   47.612499    -122.345642     

                                                         location.labeledLatLngs  \
0  [{'label': 'display', 'lat': 47.62207754974829, 'lng': -122.33725787057811}]    
1  [{'label': 'display', 'lat': 47.615631103515625, 'lng': -122.35330963134766}]   
2  [{'label': 'display', 'lat': 47.62607955932617, 'lng': -122.34188079833984}]    
3  [{'label': 'display', 'lat': 47.625269, 'lng': -122.357977}]                    
4  [{'label': 'display', 'lat': 47.61249923706055, 'lng': -122.34564208984375}]    

   location.distance location.cc location.city location.state  \
0  920                US          Issaquah      WA              
1  621                US          Seattle       WA              
2  833                US          Seattle       WA              
3  838                US          Seattle       WA              
4  933                US          Seattle       WA              

  location.country  \
0  United States     
1  United States     
2  United States     
3  United States     
4  United States     

                                    location.formattedAddress  \
0  [Mobile, Issaquah, WA]                                       
1  [2801 Western Ave, Seattle, WA 98121]                        
2  [771 Valley St, Seattle, WA]                                 
3  [Seattle, WA]                                                
4  [90 Blanchard St (btw Western and 1st), Seattle, WA 98121]   

  location.postalCode location.crossStreet delivery.id delivery.url  \
0  NaN                 NaN                  NaN         NaN           
1  98121               NaN                  NaN         NaN           
2  NaN                 NaN           

In [39]:
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
WA_dataframe_filtered = dataframe.loc[:, filtered_columns]

def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']


WA_dataframe_filtered['categories'] = WA_dataframe_filtered.apply(get_category_type, axis=1)

WA_dataframe_filtered.columns = [column.split('.')[-1] for column in WA_dataframe_filtered.columns]

WA_dataframe_filtered

name           categories  \
0  WiseGuy Italian Street Food            Food Truck            
1  Cucina De Ra Italian Restaurant        Food                  
2  Merlino Italian Baking Company         Bakery                
3  Procaccini's Family Italian Sithraunt  Other Nightlife       
4  Calozzi's Italian Kitchen              American Restaurant   
5  Scuola Italiana                        College Classroom     
6  Buca di Beppo                          Italian Restaurant    

              address        lat         lng  \
0  Mobile              47.622078 -122.337258   
1  2801 Western Ave    47.615631 -122.353310   
2  771 Valley St       47.626080 -122.341881   
3  NaN                 47.625269 -122.357977   
4  90 Blanchard St     47.612499 -122.345642   
5  NaN                 47.616707 -122.345197   
6  701 Westlake Ave N  47.625941 -122.339061   

                                                                  labeledLatLngs  \
0  [{'label': 'display', 'lat': 47.62207754974829, 'lng': -122.33725787057811}]    
1  [{'label': 'display', 'lat': 47.615631103515625, 'lng': -122.35330963134766}]   
2  [{'label': 'display', 'lat': 47.62607955932617, 'lng': -122.34188079833984}]    
3  [{'label': 'display', 'lat': 47.625269, 'lng': -122.357977}]                    
4  [{'label': 'display', 'lat': 47.61249923706055, 'lng': -122.34564208984375}]    
5  [{'label': 'display', 'lat': 47.61670683908768, 'lng': -122.34519695434977}]    
6  [{'label': 'display', 'lat': 47.62594146604359, 'lng': -122.33906149864197}]    

   distance  cc      city state        country  \
0  920       US  Issaquah  WA    United States   
1  621       US  Seattle   WA    United States   
2  833       US  Seattle   WA    United States   
3  838       US  Seattle   WA    United States   
4  933       US  Seattle   WA    United States   
5  523       US  Seattle   WA    United States   
6  977       US  Seattle   WA    United States   

                                             formattedAddress postalCode  \
0  [Mobile, Issaquah, WA]                                      NaN         
1  [2801 Western Ave, Seattle, WA 98121]                       98121       
2  [771 Valley St, Seattle, WA]                                NaN         
3  [Seattle, WA]                                               NaN         
4  [90 Blanchard St (btw Western and 1st), Seattle, WA 98121]  98121       
5  [Seattle, WA 98121]                                         98121       
6  [701 Westlake Ave N (at Broad St), Seattle, WA 98109]       98109       

           crossStreet                        id  
0  NaN                  50a2a4b8e4b0d2d4297f779c  
1  NaN                  4f3228a119836c91c7bbf2cc  
2  NaN                  4f3227cd19836c91c7bba092  
3  NaN                  582fa3ae9b615c49208a81dc  
4  btw Western and 1st  4befa9fdea570f47a59e8fd2  
5  NaN                  4c98099dd799a1cd9643af52  
6  at Broad St          40b13b00f964a5203ef61ee3

In [49]:
WA_dataframe_filtered.name

WA_venues_map = folium.Map(location=[latitude, longitude], zoom_start=13)

folium.CircleMarker(
    [latitude, longitude],
    radius=10,
    color='red',
    popup='Space Needle',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(WA_venues_map)


for lat, lng, label in zip(WA_dataframe_filtered.lat, WA_dataframe_filtered.lng, WA_dataframe_filtered.categories):
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        popup=label,
        fill = True,
        fill_color='blue',
        fill_opacity=0.6
    ).add_to(WA_venues_map)

WA_venues_map

# Boston Massachusetts - Museam of Fine Arts

In [40]:
address = '465 Huntington Avenue, Boston, Massachusetts'

geolocator = Nominatim(user_agent="foursquare_agent")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)

42.3394576 -71.09389640108918


In [41]:
search_query = 'Italian'
radius = 1000
print(search_query + ' .... Got it!')

Italian .... Got it!


In [42]:
url = 'https://api.foursquare.com/v2/venues/search?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&ll=42.3394576, -71.09389640108918&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604&query=Italian&radius=1000&limit=30'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude,ACCESS_TOKEN, VERSION, search_query, radius, LIMIT)
url

'https://api.foursquare.com/v2/venues/search?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&ll=42.3394576, -71.09389640108918&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604&query=Italian&radius=1000&limit=30'

In [43]:
results = requests.get(url).json()
results

venues = results['response']['venues']

dataframe = json_normalize(venues)
dataframe.head()

/opt/conda/envs/Python-3.7-main/lib/python3.7/site-packages/ipykernel/__main__.py:6: FutureWarning: pandas.io.json.json_normalize is deprecated, use pandas.json_normalize instead


id  \
0  4ebcaee49adf8b96f8dbaf3a   
1  5d19462f17d3b90023f58792   
2  4f322cba19836c91c7bd83c7   

                                                        name  \
0  The Italian-American Institute of Architecture of America   
1  Michael Schlow’s Italian Kitchen                            
2  Pepperoncini Italian Cafe                                   

                                                                                                                                                                                                                                       categories  \
0  []                                                                                                                                                                                                                                               
1  [{'id': '4bf58dd8d48988d110941735', 'name': 'Italian Restaurant', 'pluralName': 'Italian Restaurants', 'shortName': 'Italian', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/italian_', 'suffix': '.png'}, 'primary': True}]   
2  [{'id': '4d4b7105d754a06374d81259', 'name': 'Food', 'pluralName': 'Food', 'shortName': 'Food', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/default_', 'suffix': '.png'}, 'primary': True}]                                   

     referralId  hasPerk  location.lat  location.lng  \
0  v-1610326341  False    42.341418    -71.079656      
1  v-1610326341  False    42.344331    -71.102978      
2  v-1610326341  False    42.343504    -71.083455      

                                                                                                           location.labeledLatLngs  \
0  [{'label': 'display', 'lat': 42.341418, 'lng': -71.079656}]                                                                       
1  [{'label': 'display', 'lat': 42.344331, 'lng': -71.102978}]                                                                       
2  [{'label': 'display', 'lat': 42.343503999999996, 'lng': -71.083455}, {'label': 'entrance', 'lat': 42.343626, 'lng': -71.08351}]   

   location.distance location.postalCode location.cc location.city  \
0  1191               02118               US          Boston         
1  923                02215               US          Boston         
2  970                02115               US          Boston         

  location.state location.country               location.formattedAddress  \
0  MA             United States    [Boston, MA 02118]                       
1  MA             United States    [Boston, MA 02215]                       
2  MA             United States    [220 Huntington Ave, Boston, MA 02115]   

     location.address  
0  NaN                 
1  NaN                 
2  220 Huntington Ave

In [44]:
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
MA_dataframe_filtered = dataframe.loc[:, filtered_columns]

def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']


MA_dataframe_filtered['categories'] = MA_dataframe_filtered.apply(get_category_type, axis=1)

MA_dataframe_filtered.columns = [column.split('.')[-1] for column in MA_dataframe_filtered.columns]

MA_dataframe_filtered

name  \
0  The Italian-American Institute of Architecture of America   
1  Michael Schlow’s Italian Kitchen                            
2  Pepperoncini Italian Cafe                                   

           categories        lat        lng  \
0  None                42.341418 -71.079656   
1  Italian Restaurant  42.344331 -71.102978   
2  Food                42.343504 -71.083455   

                                                                                                                    labeledLatLngs  \
0  [{'label': 'display', 'lat': 42.341418, 'lng': -71.079656}]                                                                       
1  [{'label': 'display', 'lat': 42.344331, 'lng': -71.102978}]                                                                       
2  [{'label': 'display', 'lat': 42.343503999999996, 'lng': -71.083455}, {'label': 'entrance', 'lat': 42.343626, 'lng': -71.08351}]   

   distance postalCode  cc    city state        country  \
0  1191      02118      US  Boston  MA    United States   
1  923       02215      US  Boston  MA    United States   
2  970       02115      US  Boston  MA    United States   

                         formattedAddress             address  \
0  [Boston, MA 02118]                      NaN                  
1  [Boston, MA 02215]                      NaN                  
2  [220 Huntington Ave, Boston, MA 02115]  220 Huntington Ave   

                         id  
0  4ebcaee49adf8b96f8dbaf3a  
1  5d19462f17d3b90023f58792  
2  4f322cba19836c91c7bd83c7

In [45]:
MA_dataframe_filtered.name

MA_venues_map = folium.Map(location=[latitude, longitude], zoom_start=13)

folium.CircleMarker(
    [latitude, longitude],
    radius=10,
    color='red',
    popup='Museum of Fine Arts',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(MA_venues_map)


for lat, lng, label in zip(MA_dataframe_filtered.lat, MA_dataframe_filtered.lng, MA_dataframe_filtered.categories):
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        popup=label,
        fill = True,
        fill_color='blue',
        fill_opacity=0.6
    ).add_to(MA_venues_map)

MA_venues_map

# 4. Analysis

I will now analyze most of the restaurants that were found within the specific cities mentioned above. I will start with New York.

In [46]:
NY_venue_id = '4c40f185e26920a1d5bb5de7' #Yolanda's 
url = 'https://api.foursquare.com/v2/venues/4c40f185e26920a1d5bb5de7?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604'.format(NY_venue_id, CLIENT_ID, CLIENT_SECRET,ACCESS_TOKEN, VERSION)
url

'https://api.foursquare.com/v2/venues/4c40f185e26920a1d5bb5de7?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604'

In [52]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

7.9


This rating is decent but what about the other New York Restuarant that popped up? 

In [54]:
NY_venue_id = '4e74bc8ca809582dd59b548c'
url = 'https://api.foursquare.com/v2/venues/4c40f185e26920a1d5bb5de7?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604'.format(NY_venue_id, CLIENT_ID, CLIENT_SECRET,ACCESS_TOKEN, VERSION)
url

'https://api.foursquare.com/v2/venues/4c40f185e26920a1d5bb5de7?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604'

In [55]:
result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'price', 'hasMenu', 'likes', 'like', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'menu', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'popular', 'seasonalHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])


{'id': '4c40f185e26920a1d5bb5de7',
 'name': "Yolanda's Italian Pizzeria & Restaurant",
 'contact': {'phone': '7189932709', 'formattedPhone': '(718) 993-2709'},
 'location': {'address': '292 E 149th St',
  'crossStreet': 'off Morris Avenue',
  'lat': 40.81709844985412,
  'lng': -73.92186622574978,
  'labeledLatLngs': [{'label': 'display',
    'lat': 40.81709844985412,
    'lng': -73.92186622574978},
   {'label': 'entrance', 'lat': 40.816997, 'lng': -73.921803}],
  'postalCode': '10451',
  'cc': 'US',
  'city': 'Bronx',
  'state': 'NY',
  'country': 'United States',
  'formattedAddress': ['292 E 149th St (off Morris Avenue)',
   'Bronx, NY 10451']},
 'canonicalUrl': 'https://foursquare.com/v/yolandas-italian-pizzeria--restaurant/4c40f185e26920a1d5bb5de7',
 'categories': [{'id': '4bf58dd8d48988d1ca941735',
   'name': 'Pizza Place',
   'pluralName': 'Pizza Places',
   'shortName': 'Pizza',
   'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/pizza_',
    'suffix': '.png'},
  

In [56]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

7.9


Both venues have the same rating. That's alright because it's still a decent rating. Let's try Michigan. 

In [57]:
MI_venue_id = '4bce2b0acc8cd13aaee3c3cf'
url = 'https://api.foursquare.com/v2/venues/4c40f185e26920a1d5bb5de7?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604'.format(NY_venue_id, CLIENT_ID, CLIENT_SECRET,ACCESS_TOKEN, VERSION)
url

'https://api.foursquare.com/v2/venues/4c40f185e26920a1d5bb5de7?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604'

In [58]:
result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'price', 'hasMenu', 'likes', 'like', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'menu', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'popular', 'seasonalHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])


{'id': '4c40f185e26920a1d5bb5de7',
 'name': "Yolanda's Italian Pizzeria & Restaurant",
 'contact': {'phone': '7189932709', 'formattedPhone': '(718) 993-2709'},
 'location': {'address': '292 E 149th St',
  'crossStreet': 'off Morris Avenue',
  'lat': 40.81709844985412,
  'lng': -73.92186622574978,
  'labeledLatLngs': [{'label': 'display',
    'lat': 40.81709844985412,
    'lng': -73.92186622574978},
   {'label': 'entrance', 'lat': 40.816997, 'lng': -73.921803}],
  'postalCode': '10451',
  'cc': 'US',
  'city': 'Bronx',
  'state': 'NY',
  'country': 'United States',
  'formattedAddress': ['292 E 149th St (off Morris Avenue)',
   'Bronx, NY 10451']},
 'canonicalUrl': 'https://foursquare.com/v/yolandas-italian-pizzeria--restaurant/4c40f185e26920a1d5bb5de7',
 'categories': [{'id': '4bf58dd8d48988d1ca941735',
   'name': 'Pizza Place',
   'pluralName': 'Pizza Places',
   'shortName': 'Pizza',
   'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/pizza_',
    'suffix': '.png'},
  

In [59]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

7.9


Now let's try Houston.

In [60]:
TX_venue_id = '4abb7d4ff964a520d78320e3'
url = 'https://api.foursquare.com/v2/venues/4c40f185e26920a1d5bb5de7?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604'.format(NY_venue_id, CLIENT_ID, CLIENT_SECRET,ACCESS_TOKEN, VERSION)
url

'https://api.foursquare.com/v2/venues/4c40f185e26920a1d5bb5de7?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604'

In [61]:
result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'price', 'hasMenu', 'likes', 'like', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'menu', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'popular', 'seasonalHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])
7.9


In [62]:
TX_venue_id = '4f32962b19836c91c7e649f1'
url = 'https://api.foursquare.com/v2/venues/4c40f185e26920a1d5bb5de7?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604'.format(NY_venue_id, CLIENT_ID, CLIENT_SECRET,ACCESS_TOKEN, VERSION)
url

result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'price', 'hasMenu', 'likes', 'like', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'menu', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'popular', 'seasonalHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])
7.9


Now Seattle

In [63]:
WA_venue_id = '50a2a4b8e4b0d2d4297f779c'
url = 'https://api.foursquare.com/v2/venues/4c40f185e26920a1d5bb5de7?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604'.format(NY_venue_id, CLIENT_ID, CLIENT_SECRET,ACCESS_TOKEN, VERSION)
url

result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'price', 'hasMenu', 'likes', 'like', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'menu', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'popular', 'seasonalHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])
7.9


In [64]:
WA_venue_id = '4f3228a119836c91c7bbf2cc'
url = 'https://api.foursquare.com/v2/venues/4c40f185e26920a1d5bb5de7?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604'.format(NY_venue_id, CLIENT_ID, CLIENT_SECRET,ACCESS_TOKEN, VERSION)
url

result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'price', 'hasMenu', 'likes', 'like', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'menu', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'popular', 'seasonalHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])
7.9


In [65]:
WA_venue_id = '4f3227cd19836c91c7bba092'
url = 'https://api.foursquare.com/v2/venues/4c40f185e26920a1d5bb5de7?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604'.format(NY_venue_id, CLIENT_ID, CLIENT_SECRET,ACCESS_TOKEN, VERSION)
url

result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'price', 'hasMenu', 'likes', 'like', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'menu', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'popular', 'seasonalHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])
7.9


In [66]:
WA_venue_id = '582fa3ae9b615c49208a81dc'
url = 'https://api.foursquare.com/v2/venues/4c40f185e26920a1d5bb5de7?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604'.format(NY_venue_id, CLIENT_ID, CLIENT_SECRET,ACCESS_TOKEN, VERSION)
url

result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'price', 'hasMenu', 'likes', 'like', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'menu', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'popular', 'seasonalHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])
7.9


In [67]:
WA_venue_id = '4befa9fdea570f47a59e8fd2'
url = 'https://api.foursquare.com/v2/venues/4c40f185e26920a1d5bb5de7?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604'.format(NY_venue_id, CLIENT_ID, CLIENT_SECRET,ACCESS_TOKEN, VERSION)
url

result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'price', 'hasMenu', 'likes', 'like', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'menu', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'popular', 'seasonalHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])
7.9


And then let's do Massachusetts

In [68]:
MA_venue_id = '4ebcaee49adf8b96f8dbaf3a'
url = 'https://api.foursquare.com/v2/venues/4c40f185e26920a1d5bb5de7?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604'.format(NY_venue_id, CLIENT_ID, CLIENT_SECRET,ACCESS_TOKEN, VERSION)
url

result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'price', 'hasMenu', 'likes', 'like', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'menu', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'popular', 'seasonalHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])
7.9


In [69]:
MA_venue_id = '5d19462f17d3b90023f58792'
url = 'https://api.foursquare.com/v2/venues/4c40f185e26920a1d5bb5de7?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604'.format(NY_venue_id, CLIENT_ID, CLIENT_SECRET,ACCESS_TOKEN, VERSION)
url

result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'price', 'hasMenu', 'likes', 'like', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'menu', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'popular', 'seasonalHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])
7.9


In [70]:
MA_venue_id = '4f322cba19836c91c7bd83c7'
url = 'https://api.foursquare.com/v2/venues/4c40f185e26920a1d5bb5de7?client_id=BVHVVQMLE03WWNN1YMOZZ5BBT4SGCVWQA3Y2DGVA44ZBXLUJ&client_secret=GL12QT0SOAFRYPELNEH0RUTO3J0DGXKYSPYSLVAEE5Y4ZQ3I&oauth_token=BNNWRWLRJIQ31AVOXGYDOCCERWVBLU5CPOF2CTS0JFKTV3WV&v=20180604'.format(NY_venue_id, CLIENT_ID, CLIENT_SECRET,ACCESS_TOKEN, VERSION)
url

result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'price', 'hasMenu', 'likes', 'like', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'menu', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'popular', 'seasonalHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])
7.9


# 5. and 6. Results and Discussion and Conclusion

While the results of searching for Italian restuarants came back rather well, it would appear that the analysis of each of the restuarants popularity were rather inconclusive since all of them cam back as 7.9. 

This is a decent rating for the restaurants and would definitely warrant a sit down to eat at. However, the original problem of "where is the best Italian restuarant in these cities" still remains somewhat unanswered. 

A few ways to perhaps improve this study would be the following:

1. Expand the radius 
2. Expand the type of cuisine to include more options
3. Include more information on the restaurant, such as item menus, reservations required, etc. 
